client/finality-grandpa/src/communication/tests.rs

```diff
@@ -44,11 +43,19 @@ struct TestNetwork {
 	sender: mpsc::UnboundedSender<Event>,
 }
 
-impl sc_network_gossip::Network<Block> for TestNetwork {
-	fn event_stream(&self) -> Box<dyn futures::Stream<Item = NetworkEvent, Error = ()> + Send> {
+impl TestNetwork {
+	fn event_stream_03(&self) -> Pin<Box<dyn futures::Stream<Item = NetworkEvent> + Send>> {
```

Comment:
Why this indirection through `event_stream_03`? Why not do this within `sc_network_gossip::Network::event_stream`?

Reply:
Yeah, that's a little wierd, but It's gonna get changed via https://github.com/paritytech/substrate/pull/4633 which removed pretty much all the remaining futures01 stuff.